### PR TITLE
[Mono.Debugger.Soft] Properly handle InvalidStackFrameException withi…

### PR DESCRIPTION
**Pull request by jstedfast**
…n StackFrame.Domain

### CODE REVIEW

File: mcs/class/Mono.Debugger.Soft/Mono.Debugger.Soft/StackFrame.cs [hunk 1]
```diff
@@ -39,6 +39,8 @@ public class StackFrame : Mirror
 					if (vm.Version.AtLeast (2, 38)) {
 						try {
 							domain = vm.GetDomain (vm.conn.StackFrame_GetDomain (thread.Id, Id));
+						} catch (InvalidStackFrameException) {
+							domain = Thread.Domain;
```
Comment (vargaz) [inline mcs/class/Mono.Debugger.Soft/Mono.Debugger.Soft/StackFrame.cs]: This is just masking an underlying problem.

Comment (jstedfast) [reply mcs/class/Mono.Debugger.Soft/Mono.Debugger.Soft/StackFrame.cs]: I know, but it also makes the behavior consistent with vm versions < 2.38 when the stack frame is invalid.

**Comment by jstedfast** [comment]
Guys, a big customer has been waiting nearly a year for a fix to this... please merge this or come up with a better fix ASAP.

**Comment by thaystg** [comment]
Hey, @jstedfast, do you know how to reproduce?

**Comment by jstedfast** [comment]
I wish I knew, but I've been unable to get a test case and have been unable to reproduce locally myself, so all I have to go on is the exceptions in the logs.

**Comment by thaystg** [comment]
I think we can close this and wait if it happens again after this fix: https://github.com/mono/debugger-libs/pull/286

**Comment by jstedfast** [comment]
sounds good to me, thanks for looking into a real fix for this issue @thaystg !

**Pull request closed without merging**